File: C++/Program/Car.cpp
/* 
    Saya Adrian Mulianto NIM [2200939] mengerjakan soal Latihan Praktikum-2 dalam mata kuliah DPBO
    untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin 
*/

// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>
#include "Vehicle.cpp"

// Deklarasi namespace
using namespace std;

/*
    Kelas Car yang digunakan untuk merepresentasikan data Car
*/
class Car : public Vehicle{
private:
    // Atribut yang digunakan pada kelas Car terdiri dari merk Car beserta jenis prosesornya
    string jumlah_kursi;
    string jumlah_pintu;
public:
    // Constructor
    Car(){

    }

    string getType() const override {
        return "Car";
    }

    // Constructor : Overloading
    Car(string jumlah_kursi, string plat_nomor, string merek, string tahun_produksi, string warna) : Vehicle(plat_nomor, merek, tahun_produksi, warna){
        this->jumlah_kursi = jumlah_kursi;
    }

    // Constructor : Overloading
    Car(string jumlah_kursi, string jumlah_pintu, string plat_nomor, string merek, string tahun_produksi, string warna) : Vehicle(plat_nomor, merek, tahun_produksi, warna){
        this->jumlah_kursi = jumlah_kursi;
        this->jumlah_pintu = jumlah_pintu;
    }

    // Method setter untuk set nilai setiap atribut pada kelas Car
    void setjumlah_kursi(string jumlah_kursi){
        this->jumlah_kursi = jumlah_kursi;
    }

    void setjumlah_pintu(string jumlah_pintu){
        this->jumlah_pintu = jumlah_pintu;
    }

    // Method getter untuk mendapatkan nilai setiap atribut pada kelas Car
    string getjumlah_kursi(){
        return jumlah_kursi;
    }

    string getjumlah_pintu(){
        return jumlah_pintu;
    }

    // Destructor
    ~Car(){

    }
};

File: Python/Program/Car.py

    # Saya Adrian Mulianto NIM [2200939] mengerjakan soal Latihan Praktikum-4 dalam mata kuliah DPBO
    # untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin 

from Vehicle import Vehicle

class Car(Vehicle):
    def __init__(self, plat_nomor = "", merek = "", tahun_produksi = "", warna = "", jumlah_kursi = "", jumlah_pintu = ""):
        super().__init__(plat_nomor, merek, tahun_produksi, warna)
        self.jumlah_kursi = jumlah_kursi
        self.jumlah_pintu = jumlah_pintu

    def get_type(self):
        return "Car"

    # Method setter untuk set nilai setiap atribut pada kelas car
    def set_jumlah_kursi(self, jumlah_kursi):
        self.jumlah_kursi = jumlah_kursi

    def set_jumlah_pintu(self, jumlah_pintu):
        self.jumlah_pintu = jumlah_pintu

    # Method getter untuk mendapatkan nilai setiap atribut pada kelas car
    def get_jumlah_kursi(self):
        return self.jumlah_kursi

    def get_jumlah_pintu(self):
        return self.jumlah_pintu